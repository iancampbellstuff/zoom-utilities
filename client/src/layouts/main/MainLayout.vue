<template>
    <q-layout id="main-layout" view="lHh Lpr lFf">
        <q-header elevated>
            <q-toolbar>
                <q-toolbar-title> Zoom Utilities </q-toolbar-title>
                <q-tabs v-model="tabName">
                    <q-route-tab
                        v-for="t in tabs"
                        :icon="t.icon"
                        :key="t.name"
                        :label="t.label"
                        :name="t.name"
                        :to="t.route"
                        @click="onTabClick(t.name)"
                        no-caps
                        exact
                    />
                    <q-tab
                        icon="clear"
                        label="Clear All"
                        name="clearAll"
                        @click="onClearAllClick"
                        no-caps
                    />
                </q-tabs>
            </q-toolbar>
        </q-header>
        <q-page-container>
            <router-view />
        </q-page-container>
        <q-footer id="footer" elevated>
            <q-toolbar>
                <q-toolbar-title id="footer-text">
                    <a
                        title="https://github.com/iancampbellstuff"
                        href="https://github.com/iancampbellstuff"
                        >https://github.com/iancampbellstuff</a
                    >
                </q-toolbar-title>
            </q-toolbar>
        </q-footer>
    </q-layout>
</template>

<style scoped lang="scss">
#main-layout {
    #footer {
        text-align: center;
        #footer-text {
            font-size: 90%;
        }
    }
}
</style>

<script setup lang="ts">
import { onMounted, ref } from 'vue';
import { useRoute, useRouter } from 'vue-router';
import { useMeetingsStore, usePasscodeStore } from '../../stores';
import { tabs } from './mainTabs';
import { toast } from '../../utils';

const router = useRouter();
const meetingsStore = useMeetingsStore();
const passcodeStore = usePasscodeStore();
const tabName = ref(tabs[0].name);
const path = ref<string>();

onMounted(() => {
    const route = useRoute();
    path.value = route.path;
});
const onClearAllClick = () => {
    meetingsStore.resetAll();
    passcodeStore.resetAll();
    const { value } = path;
    if (value !== '/' && value !== '/home') {
        //TODO: Promises must be handled appropriately or explicitly marked as ignored with the `void` operator.eslint@typescript-eslint/no-floating-promises
        void router.push('home');
    }
    toast('All fields have been cleared.');
};
const onTabClick = (tabName: string) => {
    tabName = tabName;
};
</script>
