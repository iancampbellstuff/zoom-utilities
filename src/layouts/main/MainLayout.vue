<template>
    <q-layout id="main-layout" view="lHh Lpr lFf">
        <q-header elevated>
            <q-toolbar>
                <q-toolbar-title>
                    Zoom Utilities
                </q-toolbar-title>
                <q-tabs v-model="tabName">
                    <q-route-tab
                        v-for="t in tabs"
                        :icon="t.icon"
                        :key="t.name"
                        :label="t.label"
                        :name="t.name"
                        :to="t.route"
                        @click="onTabClick(t.name)"
                        no-caps
                        exact
                    />
                    <q-tab
                        icon="clear"
                        label="Clear All"
                        name="clearAll"
                        @click="onClearAllClick"
                        no-caps
                    />
                </q-tabs>
            </q-toolbar>
        </q-header>
        <q-page-container>
            <router-view />
        </q-page-container>
        <q-footer id="footer" elevated>
            <q-toolbar>
                <q-toolbar-title id="footer-text">
                    <a
                        title="https://github.com/icampbell2"
                        href="https://github.com/icampbell2"
                        >https://github.com/icampbell2</a
                    >
                </q-toolbar-title>
            </q-toolbar>
        </q-footer>
    </q-layout>
</template>

<style lang="scss" scoped>
#main-layout {
    #footer {
        text-align: center;
        #footer-text {
            font-size: 90%;
        }
    }
}
</style>

<script lang="ts">
import Vue from 'vue';
import { toast } from '../../utils';
import { ITab, tabs } from './mainTabs';
interface IData {
    tabName: string;
    tabs: ITab[];
}
export default Vue.extend({
    name: 'MainLayout',
    data(): IData {
        return {
            tabName: '',
            tabs
        };
    },
    methods: {
        onClearAllClick() {
            this.$store.commit('passcodeModule/resetAll');
            const { path } = this.$router.currentRoute;
            if (path !== '/' && path !== '/home') {
                //TODO: Promises must be handled appropriately or explicitly marked as ignored with the `void` operator.eslint@typescript-eslint/no-floating-promises
                void this.$router.push('home');
            }
            toast('All fields have been cleared.');
        },
        onTabClick(tabName: string) {
            this.tabName = tabName;
        }
    }
});
</script>
