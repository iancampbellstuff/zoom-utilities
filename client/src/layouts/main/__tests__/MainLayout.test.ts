import { mount, VueWrapper } from '@vue/test-utils';
import { createTestingPinia } from '@pinia/testing';
import Quasar from 'quasar';
import MainLayout from '../MainLayout.vue';

describe('MainLayout', () => {
    let wrapper: VueWrapper<any>;
    beforeEach(() => {
        wrapper = mount(MainLayout, {
            global: {
                plugins: [Quasar, createTestingPinia()]
            },
            stubs: [
                'q-btn-group',
                'q-btn',
                'q-footer',
                'q-form',
                'q-header',
                'q-icon',
                'q-input',
                'q-layout',
                'q-page-container',
                'q-page',
                'q-route-tab',
                'q-tab',
                'q-tabs',
                'q-toolbar-title',
                'q-toolbar',
                'q-tooltip',
                'router-view'
            ]
        });
    });
    test('is a Vue instance', () => {
        expect(wrapper.vm).toBeTruthy();
    });
    it('should render correctly', () => {
        expect(wrapper.element).toMatchSnapshot();
    });
});
