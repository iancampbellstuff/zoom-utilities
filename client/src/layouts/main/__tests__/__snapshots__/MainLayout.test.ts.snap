// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainLayout should render correctly 1`] = `
<q-layout
  id="main-layout"
  view="lHh Lpr lFf"
>
  <q-header
    elevated=""
  >
    <q-toolbar>
      <q-toolbar-title>
        Zoom Utilities
      </q-toolbar-title>
      <q-tabs
        modelvalue="home"
      >
        
        <q-route-tab
          exact=""
          icon="home"
          label="Home"
          name="home"
          no-caps=""
          to="/home"
        />
        <q-route-tab
          exact=""
          icon="lock"
          label="Passcode"
          name="passcode"
          no-caps=""
          to="/passcode"
        />
        <q-route-tab
          exact=""
          icon="groups"
          label="Meetings"
          name="meetings"
          no-caps=""
          to="/meetings"
        />
        <q-route-tab
          exact=""
          icon="videocam"
          label="Recordings"
          name="recordings"
          no-caps=""
          to="/recordings"
        />
        <q-route-tab
          exact=""
          icon="login"
          label="Accounts"
          name="accounts"
          no-caps=""
          to="/accounts"
        />
        
        <q-tab
          icon="clear"
          label="Clear All"
          name="clearAll"
          no-caps=""
        />
      </q-tabs>
    </q-toolbar>
  </q-header>
  <q-page-container>
    <q-page
      class="row items-center justify-evenly"
      id="index"
    >
      <img
        alt="Zoom Utilities"
        id="logo"
        src=""
      />
    </q-page>
  </q-page-container>
  <q-footer
    elevated=""
    id="footer"
  >
    <q-toolbar>
      <q-toolbar-title
        id="footer-text"
      >
        <a
          href="https://github.com/iancampbellstuff"
          title="https://github.com/iancampbellstuff"
        >
          https://github.com/iancampbellstuff
        </a>
      </q-toolbar-title>
    </q-toolbar>
  </q-footer>
</q-layout>
`;
